skimage/measure/fit.py

```diff
@@ -680,6 +747,30 @@ def _dynamic_max_trials(n_inliers, n_samples, min_samples, probability):
     return np.ceil(np.log(nom) / np.log(denom))
 
 
+def add_from_estimate(cls):
+    """Add ``from_estimate`` method  class using ``estimate`` method"""
+
+    if hasattr(cls, 'from_estimate'):
+        return cls
+
+    warn(
+        "Passing custom classes without `from_estimate` deprecated "
+        "since version 0.26 and will be removed in version 2.2. "
+        "Add `from_estimate` method to custom class to avoid this warning.",
```

Comment:
Hmm, this doesn't mention that we expect and call `from_estimate` as a classmethod. And asking users to implement a classmethod seems a bit more advanced than asking for a simple method. 

I'm wondering if we can provide an abstract base or – long-term – an [protocol](https://docs.python.org/3/library/typing.html#typing.Protocol) instead. That would also make the documentation a lot easier since we could simply point them at that class.

Reply:
I've noted that `from_estimate` should be a class method.   For the protocol - yes - we could do that - but given in this case we only need two things (`from_estimate` and `residuals`), and given that making a custom class will be somewhat advanced, perhaps we can just extend the docstring a bit to improve the documentation?

Reply:
I don't follow? Providing an ABC, pointing at it and saying "hey this is the API we expect from `model_class=`" seems a lot simpler than explaining that in a docstring?

But I'm okay with giving the docstring a shot. :+1:

Reply:
I'm not against an ABC - only saying that once you've got to the level of the kind of user that would benefit from an ABC for this two-method API, they probably won't be put off by the instruction to define a class method.

Reply:
OK - I added a `Protocol`.  It turns out you can't (as far as I can see) use Protocol to require a method to be a class method, so I've added a separate check for that, and added tests for protocol violations.